src/hpc-cache/azext_hpc_cache/_params.py

```diff
@@ -32,102 +39,80 @@ def load_arguments(self, _):
         c.argument('sku_name', help='SKU name for this Cache.', required=True)
 
     with self.argument_context('hpc-cache update') as c:
-        c.argument('resource_group_name', resource_group_name_type)
         c.argument('name', help='Name of Cache.')
         c.argument('tags', tags_type)
-        c.argument('location', arg_type=get_location_type(self.cli_ctx))
-        c.argument('cache_size_gb', help='The size of this Cache, in GB.')
-        c.argument('subnet', help='Subnet used for the Cache.')
-        c.argument('sku_name', help='SKU name for this Cache.')
+        c.argument('location', arg_type=get_location_type(self.cli_ctx), deprecate_info=c.deprecate(hide=True))
+        c.argument('cache_size_gb', help='The size of this Cache, in GB.', deprecate_info=c.deprecate(hide=True))
+        c.argument('subnet', help='Subnet used for the Cache.', deprecate_info=c.deprecate(hide=True))
+        c.argument('sku_name', help='SKU name for this Cache.', deprecate_info=c.deprecate(hide=True))
 
     with self.argument_context('hpc-cache delete') as c:
-        c.argument('resource_group_name', resource_group_name_type)
         c.argument('name', help='Name of Cache.')
 
     with self.argument_context('hpc-cache show') as c:
-        c.argument('resource_group_name', resource_group_name_type)
         c.argument('name', help='Name of Cache.')
 
     with self.argument_context('hpc-cache wait') as c:
-        c.argument('resource_group_name', resource_group_name_type)
-        c.extra('name', help='Name of Cache.', validator=transfer_cache_name, required=True)
-        c.ignore('cache_name')
+        c.argument('cache_name', options_list=['--name'], help='Name of Cache.', required=True)

```

Comment:
Define a CLIArgumentType for "cache name" for reuse.

Reply:
As for the definition of argument cache name:

Most are:
```
c.argument('cache_name', help='Name of Cache.')
```
Some are:
```
c.argument('name', help='Name of Cache.')
```
Another one is:
```
c.argument('cache_name', options_list=['--name', '-n'], help='Name of Cache.', required=True)
```

It seems that the only part that can be reused is `help='Name of Cache.'`. May I ask is it necessary to define type in this case?